core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -234,17 +208,15 @@ package internal {
       super.generateComponent()
     }
 
-    private[chisel3] override def initializeInParent(parentCompileOptions: CompileOptions): Unit = {
-      // Don't generate source info referencing parents inside a module, since this interferes with
-      // module de-duplication in FIRRTL emission.
-      implicit val sourceInfo = UnlocatableSourceInfo
-
-      if (!parentCompileOptions.explicitInvalidate) {
-        pushCommand(DefInvalid(sourceInfo, _io.ref))
+    override def _compatAutoWrapPorts(): Unit = {
```

Comment:
Does this change behavior, that LegacyModule does auto-wrapping instead of just Chisel.Module? Is this intended? And if so, what should the difference between the two be, or can they be mostly merged?

Reply:
> Does this change behavior, that LegacyModule does auto-wrapping instead of just Chisel.Module? Is this intended?

Yes that's right, and it is intentional. I thought it better that the reflection logic live inside of `package chisel3` so that `lazy val _io` could be private rather than leak (since `package object Chisel` is not in package `chisel3` so we can't use package private). I'll agree it's a bit awkward if anyone is using `LegacyModule`.

The alternative I thought about was adding trait in `package chisel3.internal` that extends `LegacyModule` with this new functionality. I ended up folding it into `LegacyModule` because it felt like just extra levels of inheritance that didn't actually do anything useful, and as `LegacyModule` is already in package `chisel.internal`, you're not supposed to be using it. That being said people *could be* so we could consider deprecating inheritance from `LegacyModule`, adding the extra layer of hierarchy, and folding them together in the next version. I just decided to keep it simpler.

> And if so, what should the difference between the two be, or can they be mostly merged?

The difference is minimal, really just the addition of the constructors allowing for overriding `clock` and `reset` in `Chisel.Module`. They could be merged, although doing so inside of package `chisel.internal` and using a type alias would harm the ScalaDoc of `Chisel.Module`. I don't feel too strongly either way though.

Reply:
I think it might be cleaner to move all the code into LegacyModule (so Chisel.Module is just a type alias) so we don't imply that there should be some kind of difference. If the documentation doesn't resolve cleanly, we can always make Chisel.Module an empty extension of LegacyModule (and document the rationale in the docs)?

Reply:
It turns out that `Chisel.Module` is already a type alias for `Chisel.CompatibilityModule` so there's no harm in folding it into `chisel3.internal.LegacyModule` 🤷, I'll go ahead and do that and update the ScalaDoc

Reply:
On the flip side, we exclude everything in package `chisel3.internal` from the ScalaDoc, so with `Chisel.Module` as a type alias for `chisel3.internal.LegacyModule`, there is no ScalaDoc :-/

Reply:
Done, I've changed `Chisel.Module` to be a type alias for `chisel3.internal.LegacyModule`.